packages/plugin-core/src/_extension.ts

```diff
@@ -553,8 +554,22 @@ export async function _activate(
     ws.workspaceImpl = undefined;
 
     const currentVersion = DendronExtension.version();
-    const previousWorkspaceVersion = stateService.getWorkspaceVersion();
-    const previousGlobalVersion = stateService.getGlobalVersion();
+
+    // this used to be handled by StateService (if undefined, 0.0.0).
+    // this will be reassigned if we are in a dendron workspace.
+    let previousWorkspaceVersion = "0.0.0";
+
+    const previousGlobalVersionFromState = stateService.getGlobalVersion();
+
+    // temporarily here to backfill globalState into metadata
+    // this should be removed once we determine that
+    // we have sufficiently backfilled out user.
+    if (previousGlobalVersionFromState) {
```

Comment:
won't users that upgraded recently also have a version that is already written? should this logic change so that if no current global version is found in `metadataService`, then try to look for it in state service?

Reply:
in `showWelcomeOrWhatsNew`, we call `metadataService.setGlobalVersion(version)` with the current extension version.

Everyone encountering this bug would have already have the global version written in metadata. It's just that we didn't read that during initialization thus `IntallStatus` was set to `UPGRADED`.